GenumKit/Parsers/AssetsCatalogParser.swift

```diff
@@ -20,14 +20,85 @@ public final class AssetsCatalogParser {
     }
   }
 
-  public func parseDirectory(path: String) {
-    if let dirEnum = NSFileManager.defaultManager().enumeratorAtPath(path) {
-      while let path = dirEnum.nextObject() as? NSString {
-        if path.pathExtension == "imageset" {
-          let imageName = (path.lastPathComponent as NSString).stringByDeletingPathExtension
-          self.addImageName(imageName)
+  public func parseCatalog(path: String) {
+    guard let items = loadAssetCatalogContents(path) else { return }
+
+    // process recursively
+    process(items)
+  }
+}
+
+// Mark: - Plist processing
+
+private enum AssetCatalog: String {
+  case children = "children"
+  case filename = "filename"
+  case providesNamespace = "provides-namespace"
+  case root = "com.apple.actool.catalog-contents"
+}
+
+extension AssetsCatalogParser {
+  static let ImageSet = "imageset"
+
+  private func process(items: [[String: AnyObject]], prefix: String = "") {
+    for item in items {
+      let filename = item[AssetCatalog.filename.rawValue] as! NSString
+
+      // this is a simple imageset
+      if filename.pathExtension == AssetsCatalogParser.ImageSet {
```

Comment:
SwiftGen has a dependency to `PathKit` which is a pod providing Swiftier API for manipulating files, file extensions, basenames, etc. 
You should switch your code so it uses it (instead of having to convert to NSString and use Foundation) 😉